presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -449,7 +454,7 @@ public IPrestoSparkQueryExecution create(
                                 warningCollector));
 
                 log.info(textDistributedPlan(fragmentedPlan, metadata.getFunctionAndTypeManager(), session, true));
-                fragmentedPlan = configureOutputPartitioning(session, fragmentedPlan);
+                fragmentedPlan = configureOutputPartitioning(session, fragmentedPlan, planAndMore.getPhysicalResourceSettings());
```

Comment:
Instead of passing `physicalResourceSettings`, you can directly determine the final hash_partition_count using:
```
(physicalResourceSettings.isValid()) ? physicalResourceSettings.getShufflePartitionCount() : getHashPartitionCount(session)
```

and then just pass the hash_partition_count value to `configureOutputPartitioning` method.